middleware/auth.go

```diff
@@ -0,0 +1,32 @@
+package middleware
+
+import (
+	"net/http"
+	"strings"
+)
+
+func isAuthorized(r *http.Request) bool {
+	// TODO: Add configuration to enable this Auth
+	// This configuration should not be as command line option
+	// Because, one glusterd can be run with `--no-auth` and
+	// other glusterd with auth enabled.
+	if strings.HasPrefix(r.RemoteAddr, "127.0.0.1:") {
```

Comment:
r.RemoteAddr pattern is like <host>:<random_port>(Not 24007, may be multiple go routine gets different port).

- r.RemoteAddr data type is string, which includes host name and port information. String comparision is one step compared to `net.SplitHostPort(ip)`, `net.ParseIP(host)` and `ip.Equal()`
- Also this field is not set by the user, no need to validate that it is valid IP or not, we just have to compare with existing pattern.
- We are not resolving IP here, so not finding enough benefit to use net.* functions for comparison.